src/sigutils/lib.c

```diff
@@ -126,6 +128,30 @@ su_lib_fftw_strategy(void)
   return su_measure_ffts ? FFTW_MEASURE : FFTW_ESTIMATE;
 }
 
+SU_FFTW(_plan)
+su_lib_plan_dft_1d(int n, SU_FFTW(_complex) *in, SU_FFTW(_complex) *out,
+        int sign, unsigned flags)
+{
+  SU_FFTW(_plan) plan;
+  int nthreads;
+
+  if (n < 32768)
+    nthreads = 1;
+  else if (n == 32768)
+    nthreads = 2;
+  else
+    nthreads = 4;
+
+  if (pthread_mutex_lock(&fft_plan_mutex) == -1)
+    return NULL;
+  SU_FFTW(_plan_with_nthreads)(nthreads);
+  plan = SU_FFTW(_plan_dft_1d)(n, in, out, sign, flags);
+  SU_FFTW(_plan_with_nthreads)(1);
+  pthread_mutex_unlock(&fft_plan_mutex);
```

Comment:
Next time consider prefixing global variables by `g_`. While there are still a bunch of variables that do not follow this convention, it helps me a lot when I am reading new code.